src/engine.rs

```diff
@@ -110,14 +103,15 @@ impl Engine {
             ..
         } = set;
 
-        let blocker_options = BlockerOptions {
-            enable_optimizations: optimize,
-        };
+        let memory = FlatBufferBuilder::make_flatbuffer(network_filters, optimize);
+
+        let filter_data_context = FilterDataContext::new(memory);
 
         Self {
-            blocker: Blocker::new(network_filters, &blocker_options),
+            blocker: Blocker::from_context(filter_data_context.clone()),
```

Comment:
Convention is to explicitly use `Rc::clone(x)` or `Arc::clone(x)` rather than `x.clone()` to make it clear that it's just a refcount increase.

https://doc.rust-lang.org/book/ch15-04-rc.html#using-rct-to-share-data

In these instances it depends on the feature config, so I'd suggest `FilterDataContextRef::clone(x)` instead